mmv1/third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb

```diff
@@ -169,24 +313,71 @@ func resourceDataflowFlexTemplateJobRead(d *schema.ResourceData, meta interface{
 		return handleNotFoundError(err, d, fmt.Sprintf("Dataflow job %s", jobId))
 	}
 
+	if err := d.Set("job_id", job.Id); err != nil {
+		return fmt.Errorf("Error setting job_id: %s", err)
+	}
 	if err := d.Set("state", job.CurrentState); err != nil {
 		return fmt.Errorf("Error setting state: %s", err)
 	}
 	if err := d.Set("name", job.Name); err != nil {
 		return fmt.Errorf("Error setting name: %s", err)
 	}
+	if err := d.Set("type", job.Type); err != nil {
+		return fmt.Errorf("Error setting type: %s", err)
+	}
 	if err := d.Set("project", project); err != nil {
 		return fmt.Errorf("Error setting project: %s", err)
 	}
 	if err := d.Set("labels", job.Labels); err != nil {
 		return fmt.Errorf("Error setting labels: %s", err)
 	}
+	if err := d.Set("kms_key_name", job.Environment.ServiceKmsKeyName); err != nil {
+		return fmt.Errorf("Error setting kms_key_name: %s", err)
+	}
+	if err := d.Set("service_account_email", job.Environment.ServiceAccountEmail); err != nil {
+		return fmt.Errorf("Error setting service_account_email: %s", err)
+	}
 
-  if ok := shouldStopDataflowJobDeleteQuery(job.CurrentState, d.Get("skip_wait_on_job_termination").(bool)); ok {
+	sdkPipelineOptions, err := ConvertToMap(job.Environment.SdkPipelineOptions)
+	if err != nil {
+		return err
+	}
+	optionsMap := sdkPipelineOptions["options"].(map[string]interface{})
+
+	if err := d.Set("temp_location", optionsMap["tempLocation"]); err != nil {
+		return fmt.Errorf("Error setting temp_gcs_location: %s", err)
+	}
+	if err := d.Set("network", optionsMap["network"]); err != nil {
+		return fmt.Errorf("Error setting network: %s", err)
+	}
+	if err := d.Set("num_workers", optionsMap["numWorkers"]); err != nil {
+		return fmt.Errorf("Error setting num_workers: %s", err)
+	}
+	if err := d.Set("max_workers", optionsMap["maxWorkers"]); err != nil {
+		return fmt.Errorf("Error setting max_workers: %s", err)
+	}
+	if err := d.Set("staging_location", optionsMap["stagingLocation"]); err != nil {
+		return fmt.Errorf("Error setting staging_location: %s", err)
+	}
+	if err := d.Set("sdk_container_image", optionsMap["sdkContainerImage"]); err != nil {
+		return fmt.Errorf("Error setting sdk_container_image: %s", err)
+	}
+	if err := d.Set("network", optionsMap["network"]); err != nil {
+		return fmt.Errorf("Error setting network: %s", err)
+	}
+	if err := d.Set("subnetwork", optionsMap["subnetwork"]); err != nil {
+		return fmt.Errorf("Error setting subnetwork: %s", err)
+	}
+	if err := d.Set("machine_type", optionsMap["workerMachineType"]); err != nil {
+		return fmt.Errorf("Error setting machine_type: %s", err)
+	}
+
+	if ok := shouldStopDataflowJobDeleteQuery(job.CurrentState, d.Get("skip_wait_on_job_termination").(bool)); ok {
 		log.Printf("[DEBUG] Removing resource '%s' because it is in state %s.\n", job.Name, job.CurrentState)
 		d.SetId("")
 		return nil
 	}
+	d.SetId(job.Id)
```

Comment:
I'm not sure what the behavior would have been previously without setting this, but I believe this change could result in unexpected behavior for existing users.

Reply:
Removed this line, I did not realize it is a duplicate, It is already set both in create and update functions as expected